### PR TITLE
Allow creating jobs

### PR DESCRIPTION
**Pull request by AdiZav**
##### Short description:
Allow creating jobs with minimal config

### CODE REVIEW

**Review by myakove — DISMISSED** [review]

**Review by sarahbx — APPROVED** [review]

File: ocp_resources/job.py [hunk 1]
```diff
@@ -7,3 +8,57 @@ class Job(NamespacedResource):
...
+                self.res["spec"]["template"]["spec"][
+                    "serviceAccount"
+                ] = self.service_account
+
+            if self.restart_policy:
+                self.res["spec"]["template"]["spec"][
+                    "restartPolicy"
+                ] = self.restart_policy
+
+            self.res["spec"]["template"]["spec"]["containers"] = self.containers
```
Comment (rnetser) [inline ocp_resources/job.py:62]: as this always happens, for readability - move under the definition of template.spec.

**Comment by AdiZav** [comment]
!verifired

File: ocp_resources/job.py [hunk 2]
```diff
@@ -7,3 +8,56 @@ class Job(NamespacedResource):
...
+        self.restart_policy = restart_policy
+        self.backoff_limit = backoff_limit
+        self.service_account = service_account
+        self.containers = containers
+
+    def to_dict(self):
+        self.res = super().to_dict()
+        self.res.setdefault("spec", {})
+
+        if self.backoff_limit or self.backoff_limit == 0:
```
Comment (myakove) [inline ocp_resources/job.py:46]:
```suggestion
        if self.backoff_limit is not None:
```

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by AdiZav** [comment]
!verified

File: ocp_resources/job.py (hunk 2 above)
Comment (AdiZav) [reply ocp_resources/job.py]: Resolved

**Review by myakove — APPROVED** [review]

**Pull request merged**